ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -194,6 +197,53 @@ def _llmobs_set_meta_tags_from_chain(
             except TypeError:
                 log.warning("Failed to serialize chain output data to JSON")
 
+    def _llmobs_set_meta_tags_from_similarity_search(
+        self,
+        span: Span,
+        input_query: str,
+        output_documents: Union[List[Any], None],
+        error: bool = False,
+        is_workflow: bool = False,
+    ) -> None:
+        span.set_tag_str(SPAN_KIND, "workflow" if is_workflow else "retrieval")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, span.get_tag(PROVIDER) or "")
+
+        input_tag_key = INPUT_VALUE if is_workflow else INPUT_MESSAGES
```

Comment:
No need to use the `input.messages` field for retrievals, take a look at this document for retrieval span specifications: https://docs.google.com/document/d/1bsssupQyLq7sL-2cM_9U1R7fmIbf8CmC9cDNIOvv29k/edit?usp=sharing